Src/HALALMock/Services/EXTI/EXTI.cpp

```diff
@@ -65,5 +69,10 @@ bool ExternalInterrupt::get_pin_value(uint8_t id) {
 	}
 
 	Pin& pin = service_ids[id];
-	return HAL_GPIO_ReadPin(GPIO_PORT, pin.gpio_pin);
+	EmulatedPin &pin_data = SharedMemory::get_pin(pin);
+	if(pin_data.type != PinType::EXTIPin) {
+		ErrorHandler("ID %d is not registered as a EXTIPin",id);
+		return;
+	}
```

Comment:
I know this is probably how it is coded in the "normal" version of the STLIB, it would be interesting to make a wrapper around it so this check is only performed in debug builds / can be switched on or off at compile time.